src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -506,6 +506,9 @@ void DlgPrefWaveform::updateWaveformAcceleration(
     useAccelerationCheckBox->setEnabled(supportAcceleration &&
             supportSoftware && type != WaveformWidgetType::Empty);
 
+    // Update backend
+    slotSetWaveformAcceleration(useAccelerationCheckBox->isChecked());
+
```

Comment:
This explicit call is the fix.

Note that the block/unblock calls for `useAccelerationCheckBox` are pointless here because `slotSetWaveformAcceleration` is connected to its `clicked` signal (user interaction only), not `toggled` or `checkStateChanged` (Qt 6.7++).
I'll leave those in here, just in case someone switches to `checkStateChanged` at some point.

Reply:
Damn, was almost too easy.

While this fixes it during a session, it now chnages the waveform type after restart.
Will look into it later on..